lib/puppet/util/log.rb

```diff
@@ -284,47 +292,34 @@ def to_hash
   end
 
   def to_data_hash
-    {
+    hash = {
       'level' => @level,
       'message' => @message,
       'source' => @source,
       'tags' => @tags,
       'time' => @time.iso8601(9),
-      'file' => @file,
-      'line' => @line,
     }
+    [:backtrace, :environment, :node, :issue_code, :file, :line, :pos].each do |attr|
+      iv = "@#{attr}"
+      hash[attr.to_s] = instance_variable_get(iv) if instance_variable_defined?(iv)
+    end
+    hash
   end
 
   def to_pson(*args)
     to_data_hash.to_pson(*args)
   end
 
-  def message=(msg)
-    raise ArgumentError, "Puppet::Util::Log requires a message" unless msg
-    @message = msg.to_s
-  end
-
-  def level=(level)
```

Comment:
API change (and this one is nasty since it means you can log with high level, then lower it and still get it in the output even if it is flltered on that level)